nbt/src/main/java/net/md_5/bungee/nbt/type/StringTag.java

```diff
@@ -0,0 +1,42 @@
+package net.md_5.bungee.nbt.type;
+
+import com.google.common.base.Preconditions;
+import java.io.DataInput;
+import java.io.DataOutput;
+import java.io.IOException;
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+import net.md_5.bungee.nbt.Tag;
+import net.md_5.bungee.nbt.TypedTag;
+import net.md_5.bungee.nbt.limit.NbtLimiter;
+
+@Data
+@NoArgsConstructor
+@AllArgsConstructor
+public class StringTag implements TypedTag
+{
+    private String value;
+
+    @Override
+    public void read(DataInput input, NbtLimiter limiter) throws IOException
+    {
+        limiter.countBytes( OBJECT_HEADER + STRING_SIZE );
+        String string = input.readUTF();
+        limiter.countBytes( string.length(), Character.BYTES );
```

Comment:
I wish we could easily count bytes before parsing, but I don't see how.

Reply:
i agree, but as the len of readUTF is only a short it shouldn't be critical